### PR TITLE
Use a StringSlice flag type in `--watch-namespace`

### PR DESCRIPTION
**Pull request by mflendrich**
Follow-up to #1317.

Makes the `--watch-namespace` flag use a string slice as a storage medium which is a more robust way to accept comma-separated settings than `strings.Split(",")`
Also unrolls the namespace watch setup logic into 3 distinct cases with different behaviors, for better readability and code coverage visibility.

### CODE REVIEW

File: railgun/pkg/config/config.go [hunk 1]
```diff
@@ -136,7 +135,7 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 	flagSet.StringVar(&c.LeaderElectionID, "election-id", "5b374a9e.konghq.com", `Election id to use for status update.`)
 	flagSet.StringSliceVar(&c.FilterTags, "kong-admin-filter-tag", []string{"managed-by-ingress-controller"}, "The tag used to manage and filter entities in Kong. This flag can be specified multiple times to specify multiple tags. This setting will be silently ignored if the Kong instance has no tags support.")
 	flagSet.IntVar(&c.Concurrency, "kong-admin-concurrency", 10, "Max number of concurrent requests sent to Kong's Admin API.")
-	flagSet.StringVar(&c.WatchNamespace, "watch-namespace", corev1.NamespaceAll,
+	flagSet.StringSliceVar(&c.WatchNamespaces, "watch-namespace", nil,
```
Comment (ccfishk) [inline railgun/pkg/config/config.go:138]: nit: watch-namespaces

**Review by shaneutt — APPROVED** [review]
LGTM after Jimin's comments are resolved.

File: railgun/manager/run.go [hunk 2]
```diff
@@ -88,14 +88,19 @@ func Run(ctx context.Context, c *config.Config) error {
...
+		controllerOpts.Namespace = c.WatchNamespaces[0]
+	default:
 		// this mode does not set the Namespace option, so the manager will default to watching all namespaces
 		// MultiNamespacedCacheBuilder imposes a filter on top of that watch to retrieve scoped resources
 		// from the watched namespaces only.
-		controllerOpts.NewCache = cache.MultiNamespacedCacheBuilder(strings.Split(c.WatchNamespace, ","))
-	} else {
-		controllerOpts.Namespace = c.WatchNamespace
+		setupLog.Info("manager set up with multiple namespaces", "namespaces", c.WatchNamespace)
+		controllerOpts.NewCache = cache.MultiNamespacedCacheBuilder(c.WatchNamespaces)
```
Comment (ccfishk) [inline railgun/manager/run.go:103]: we need answer/be clear about the question:
do we watch whatever namespace user is configuring ?

File: railgun/pkg/config/config.go (hunk 1 above)
Comment (mflendrich) [reply railgun/pkg/config/config.go:138]: The name of the flag must remain `watch-namespace` for compatibility with KIC 1.x.
In 1.x, watch-namespace accepted only a single value.

I've opted for renaming the config variable name though, to underline the fact that now there can be any number of watched namespaces.

File: railgun/manager/run.go (hunk 2 above)
Comment (mflendrich) [reply railgun/manager/run.go:103]: I'm not sure that I understand the question well but I'll try to answer as accurately as I can.

KIC 2.0 gives the user a possibility to specify:
- zero namespaces; in this case, KIC should watch all namespaces
- one namespace
- multiple namespaces

If the user specifies one namespaces, we watch only a single namespace for resources.
If the user specifies more than one namespace, due to the limitations of the k8s API we watch all the namespaces, and perform filtering client-side.

**Review by rainest — APPROVED** [review]

File: railgun/manager/run.go (hunk 2 above)
Comment (ccfishk) [reply railgun/manager/run.go:103]: possibly (most cases yes), we have a namespace black list - per user requirement, we dont allow to watch those namespaces coz they are user asset. (edited) 
10:33
What I understand atm, we don't have any limitation, but we need consider about it.

**Pull request merged**